packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

```diff
@@ -282,6 +282,51 @@ void main() {
       Platform: () => fakePlatform,
     });
 
+    testUsingContext('Show current version to the upgrade message.', () async {
+      const String revision = 'abc123';
+      const String upstreamRevision = 'def456';
+      const String version = '1.2.3';
+      const String upstreamVersion = '4.5.6';
+
+      when(flutterVersion.frameworkRevision).thenReturn(revision);
+      when(flutterVersion.frameworkVersion).thenReturn(version);
+
+      final MockFlutterVersion latestVersion = MockFlutterVersion();
+
+      when(latestVersion.frameworkRevision).thenReturn(upstreamRevision);
+      when(latestVersion.frameworkVersion).thenReturn(upstreamVersion);
+
+      fakeCommandRunner.alreadyUpToDate = false;
+      fakeCommandRunner.remoteVersion = latestVersion;
+
+      processManager.addCommands(<FakeCommand>[
+        const FakeCommand(command: <String>[
+          'git', 'fetch', '--tags'
+        ]),
+        const FakeCommand(command: <String>[
+          'git', 'rev-parse', '--verify', '@{u}',
+        ],
+            stdout: upstreamRevision),
+        const FakeCommand(command: <String>[
+          'git', 'tag', '--points-at', upstreamRevision,
+        ],
+            stdout: ''),
+        const FakeCommand(command: <String>[
+          'git', 'describe', '--match', '*.*.*', '--long', '--tags', upstreamRevision,
+        ],
+            stdout: upstreamVersion),
+        const FakeCommand(command: <String>[
+          'git', 'reset', '--hard', upstreamRevision
+        ]),
+      ]);
+
+      await realCommandRunner.runCommandFirstHalf(force: true, gitTagVersion: gitTagVersion, flutterVersion: flutterVersion, testFlow: true, verifyOnly: false);
```

Comment:
Since the message you are expecting is in `runCommandFirstHalf`, it might be preferrable to run `fakeCommandRunner.runCommand` and awaiting for `FlutterCommandResult.success` instead of adding commands explicitly to `processManager`, just like other tests.

Also, please format the call as following: 
```dart
final Future<FlutterCommandResult> result = fakeCommandRunner.runCommand(
        force: false,
        continueFlow: false,
        testFlow: false,
        gitTagVersion: gitTagVersion,
        flutterVersion: flutterVersion,
        verifyOnly: true,
      );
```